### PR TITLE
Use Ignition's validation

### PR DESCRIPTION
**Pull request by ajeddeloh**
Make fuze use ignition's validation.

### CODE REVIEW

File: config/vendor/github.com/coreos/ignition/config/validate/astjson/node.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package node
```
Comment (crawford) [inline config/vendor/github.com/coreos/ignition/config/validate/astjson/node.go]: Can you break this into a separate commit?

File: config/astyaml.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package config
```
Comment (crawford) [inline config/astyaml.go]: License header

File: config/config.go [hunk 3]
```diff
@@ -27,14 +35,24 @@ func ParseAsV2_0_0(data []byte) (types.Config, error) {
...
-	if err := yaml.Unmarshal(data, &keyMap); err != nil {
-		return types.Config{}, err
-	}
+	var r report.Report
 
-	if err := assertKeysValid(keyMap, reflect.TypeOf(Config{})); err != nil {
-		return types.Config{}, err
+	nodes := yaml.UnmarshalToNode(data)
+	if nodes == nil {
+		fmt.Println("Failed to unmarshal yaml for reporting line numbers.")
```
Comment (crawford) [inline config/config.go]: I'm not a fan of these prints. This package is consumed by other applications. They should be able to print the way they like. Can we just return the report instead?

File: config/convert.go [hunk 4]
```diff
@@ -222,8 +229,13 @@ func ConvertAs2_0_0(in Config) (types.Config, error) {
 		})
 	}
 
-	if err := out.AssertValid(); err != nil {
-		return types.Config{}, err
+	r := validate.ValidateWithoutSource(reflect.ValueOf(out))
+	if r.String() != "" {
+		fmt.Println("Ignition report:")
```
Comment (crawford) [inline config/convert.go]: Can we return the report instead?

File: config/config.go (hunk 3 above)
Comment (ajeddeloh) [inline config/config.go]: Both fuze's validation and ignition's validation with return reports. Ignition's will not have line numbers since there is no corresponding json source. Should we merge the two and return that? That would lose which source it came from (although we could iterate over the report and add that to the entry messages). 

Another possibility is to refactor so this doesn't return a ignition config structure and instead just parses the yaml and returns a the fuze config structure and fuze report. It's caller can then use that result to call ConvertAs2_0_0. Then it's up to the caller to determine how the reports should be handled.

File: config/types/Passwd.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2016 CoreOS, Inc.
```
Comment (crawford) [inline config/types/Passwd.go]: lower case `passwd.go`

File: internal/main.go [hunk 6]
```diff
@@ -56,9 +56,17 @@ func main() {
...
+	stderr(report.String())
+	if report.IsFatal() {
+		stderr("Failed to parse fuze config")
+		os.Exit(1)
+	}
+
+	cfg, report := config.ConvertAs2_0_0(fuzeCfg)
+	stderr(report.String())
+	if report.IsFatal() {
+		stderr("Generated ignition config was invalid.")
```
Comment (crawford) [inline internal/main.go]: Ignition

File: config/types/storage.go [hunk 7]
```diff
@@ -15,7 +15,7 @@
 package types
 
 type Storage struct {
-	Disks       []Disk       `yaml:"disk"`
```
Comment (crawford) [inline config/types/storage.go]: Can you squash this change?

File: config/types/config.go [hunk 8]
```diff
@@ -23,7 +23,7 @@ type Config struct {
 }
 
 type Ignition struct {
-	Config IgnitionConfig
+	Config IgnitionConfig `yaml:"config"`
```
Comment (crawford) [inline config/types/config.go]: Can you squash this change?
